### ISSUE TITLE
Duplex reads have different quality of 6mA may lead to different 6mA results

### ISSUE THREAD

**Issue by juzuo-git** [nanoporetech/dorado#483]
Hi,
I running 6mA modification with R10 successfully, This is the code:

```sh
for bc in "barcode01" "barcode02"
do
        /bin/dorado basecaller \
        /Data/dna_r10.4.1_e8.2_400bps_sup@v4.2.0 \
        /${bc}_pod5/ \
        --modified-bases 6mA \
        --reference /db/col_CEN/Col-CEN_v1.2.fasta \
        > ${bc}.6mA.bam
done
```

And I carefully check the 6mA information based the tag in bam: `"MM" and "ML"`, which illustrated by this `https://samtools.github.io/hts-specs/SAMtags.pdf `
I got the position and quality of 6mA from this two tag for every long reads, but I found the reads, which are produced by both template strand (read-1a) and complement strand(read-1b) from same DNA locus , **have the completely different quality of 6mA**. 
In other words, `read-1a` and `read-1b` may be the paired reads (defined duplex by dorado?), **they have the same positions of 6mA, but quality of 6mA are totally different**.

It would lead to different results after filtering low quality of m6A (`230`, which means scaled-probabilities of 90%).
I known it may be too strict to keep same positions of 6mA for paired reads, but when I used `128` (~50%), still got many different position.

Any idea?
Looking forward to your replay.

**Comment by juzuo-git** [comment]
Never mind, I fixed it.

**Comment by tijyojwad** [comment]
@juzuo-git can you post the resolution so future readers are aware? Thank you!

**Comment by juzuo-git** [comment]
OK tijyojwad, I could explain my mistake to you, so other users could avoid this.
Due to dorado4 using the `minimap2` to map the reads, it is possible to obtain mapping multiple positions for a single read. I handled the negative `-` and positive `+` strand with the same method, and it lead to **completely different quality of 6mA** in the result of mixing up the strand information about the bam result.
When the strand information is negative `-`,  it is crucial to reverse and complement the query sequence, then the `position of 6mA` and `quality of 6mA` will be the same with `+`. Maybe you had noticed that I treated the different mapping position (same read_id) as the different read, which was incorrect.
In fact, I am going to check the real duplex reads, which should have the similar/symmetrical cigar values, to determine if `their 6mA quality` scores are approximately the same.
Furthermore, for the duplex reads, if `+` strand got methylated information about `A` on positive strand and `-` strand got information about `A` on negative strand, their 6mA quality scores will likely differ due to the variance in their respective positions.

### example:
- *red A represent the 6mA*
![图片](https://github.com/nanoporetech/dorado/assets/66865006/b803b721-3fe4-48f3-aa6f-0d909d3406f5)
- In this example, read1 and read2 are duplex reads and have different `position of 6mA`.

If I was misunderstanding the nanopore R10 duplex sequencing process, please point them out, Thx tijyojwad!

**Comment by tijyojwad** [comment]
thanks for posting the details! reverse complementing sounds like the right fix since you want to match up the same nucleotide position from both the strands.
